braindecode/models/eegdeformer.py

```diff
@@ -0,0 +1,288 @@
+    2. Linear transformation from ``in_features` to ``out_features``
+    3. Activation function
+    4. Dropout
+    5. Linear transformation from ``out_features`` back to ``in_features``
+    6. Dropout
+
+    Parameters
+    ----------
+    in_features : int
+        Size of each input sample.
+    out_features : int
+        Size of each output sample.
+    drop_prob : float, optional
+        Dropout probability. Default is `0.0`.
+    activation : nn.Module, optional
+        Activation function to apply after the first linear layer.
+        Default is `nn.GELU`.
+
+    Attributes
+    ----------
+    net : nn.Module
+        Sequential container holding the layers of the feedforward network.
+
+    """
+
+    def __init__(
+        self,
+        in_features: int,
+        out_features: int,
+        drop_prob: float = 0.0,
+        activation: nn.Module = nn.GELU,
+    ):
+        super().__init__()
+        self.net = nn.Sequential(
+            nn.LayerNorm(in_features),
+            nn.Linear(in_features=in_features, out_features=out_features),
+            activation(),
+            nn.Dropout(p=drop_prob),
+            nn.Linear(in_features=out_features, out_features=in_features),
+            nn.Dropout(p=drop_prob),
+        )
+
+    def forward(self, x):
+        return self.net(x)
+
+
+class _Attention(nn.Module):
+    def __init__(self, embed_dim, num_heads=8, dim_head=64, dropout=0.0):
+        super().__init__()
+        inner_dim = dim_head * num_heads
+        project_out = not (num_heads == 1 and dim_head == embed_dim)
+
+        self.heads = num_heads
+        self.scale = dim_head**-0.5
+
+        self.attend = nn.Softmax(dim=-1)
+        self.to_qkv = nn.Linear(embed_dim, inner_dim * 3, bias=False)
+
+        self.to_out = (
+            nn.Sequential(nn.Linear(inner_dim, embed_dim), nn.Dropout(dropout))
+            if project_out
+            else nn.Identity()
+        )
+
+    def forward(self, x):
+        qkv = self.to_qkv(x).chunk(3, dim=-1)
+        query, key, value = map(
+            lambda t: rearrange(t, "b n (h d) -> b h n d", h=self.heads), qkv
+        )
+
+        dots = torch.matmul(query, key.transpose(-1, -2)) * self.scale
+
+        attn = self.attend(dots)
+
+        out = torch.matmul(attn, value)
+        out = rearrange(out, "b h n d -> b n (h d)")
+        return self.to_out(out)
+
+
+class _Transformer(nn.Module):
+    def __init__(
+        self,
+        dim,
+        depth,
+        heads,
+        dim_head,
+        mlp_dim,
+        in_chan,
+        fine_grained_kernel=11,
+        dropout=0.0,
+    ):
+        super().__init__()
+        self.layers = nn.ModuleList([])
+        for i in range(depth):
+            dim = int(dim * 0.5)
+            self.layers.append(
+                nn.ModuleList(
+                    [
+                        _Attention(dim, heads, dim_head, dropout=dropout),
+                        _FeedForward(dim, mlp_dim, drop_prob=dropout),
```

Comment:
```suggestion
        embed_dim,
        depth,
        heads,
        dim_head,
        dim_feedforward,
        in_chan,
        fine_grained_kernel=11,
        dropout=0.0,
    ):
        super().__init__()
        self.layers = nn.ModuleList([])
        for i in range(depth):
            embed_dim= int(embed_dim* 0.5)
            self.layers.append(
                nn.ModuleList(
                    [
                        _Attention(embed_dim, heads, dim_head, dropout=dropout),
                        _FeedForward(embed_dim, dim_feedforward, drop_prob=dropout),
```
consistent naming in itself and with torch